custom_components/plugwise/config_flow.py

```diff
@@ -243,6 +244,66 @@ async def async_step_user(
         self._abort_if_unique_id_configured()
         return self.async_create_entry(title=api.smile_name, data=user_input)
 
+
+    async def async_step_reconfigure(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle reconfiguration of the integration."""
+        errors: dict[str, str] = {}
+        if user_input:
+            try:
+                api = await validate_input(self.hass, user_input)
+            except ConnectionFailedError:
+                errors[CONF_BASE] = "cannot_connect"
+            except InvalidAuthentication:
+                errors[CONF_BASE] = "invalid_auth"
+            except InvalidSetupError:
+                errors[CONF_BASE] = "invalid_setup"
+            except (InvalidXMLError, ResponseError):
+                errors[CONF_BASE] = "response_error"
+            except UnsupportedDeviceError:
+                errors[CONF_BASE] = "unsupported"
+            except Exception:  # noqa: BLE001
+                errors[CONF_BASE] = "unknown"
+            else:
+                await self.async_set_unique_id(
+                    api.smile_hostname or api.gateway_id, raise_on_progress=False
+                )
+                self._abort_if_unique_id_mismatch(reason="not_the_same_smile")
```

Comment:
What does this do? Entering a wrong Smile ID will lead to an `invalid_auth` error.

Now that I think about it some more, only when the IP-address(, username) and Smile ID of another Plugwise Gateway are entered, then that should be signaled. This line does that?

Reply:
I initially was confused by the use of MOCK_SMILE_ID in the test below.

See my feedback about this in the corresponding Core PR :)